### DISCUSSION TITLE
Parsing API Key

### DISCUSSION THREAD

**Discussion by diegotco** [sindresorhus/got#2191] [Q&A] [answered]
Hello, all:

How can I parse my API key in this code?

```
import got from 'got';

const {data} = await got.get('http://api.weatherstack.com/current', {
	json: {
		access_key: 'XXXXX....',
                 query: 'New York'
	}
}).json();

console.log(data);
```

Thanks in advance!

**Comment by izzqz** [comment, marked as answer]
You're trying to make GET request with json body. To authenticate request for weatherstak, you need to pass your `access_key` as query parameter. For got its [`searchParams`](https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#searchparams) option:

```js
import got from 'got';

const { data } = await got.get('http://api.weatherstack.com/current', {
	searchParams: {
		access_key: '***',
                query: 'New York'
	}
}).json();

console.log(data);
``` 

[ From Weatherstak API Authentication docs](https://weatherstack.com/documentation#:~:text=To%20authenticate%20with%20the%20API%2C%20simply%20use%20the%20base%20URL%20below%20and%20pass%20your%20API%20access%20key%20to%20the%20API%27s%20access_key%20parameter.)

**Reply by diegotco** [reply]
Nice, thank you very much. I was really confuse by the term _search_. I will deep dive more into it, thanks again for your help 🙌🏼

PD: I get the response by deleting those curly braces in data (`const {data} =`)
